daemon/mgr/container.go

```diff
@@ -232,46 +239,73 @@ func (mgr *ContainerManager) Restore(ctx context.Context) error {
 		}
 
 		id := container.ID
-
 		// map container's name to id.
 		mgr.NameToID.Put(container.Name, id)
 
 		// put container into cache.
 		mgr.cache.Put(id, container)
 
-		if container.State.Status != types.StatusRunning &&
-			container.State.Status != types.StatusPaused {
-			return nil
-		}
+		return nil
+	}
+
+	return mgr.Store.ForEach(fn)
+}
+
+// Restore tries to recover those alive containers
+func (mgr *ContainerManager) Restore(ctx context.Context) error {
+	// get all running containers
+	containers, err := mgr.List(ctx,
+		&ContainerListOption{
+			All: true,
+			FilterFunc: func(c *Container) bool {
+				return (c.IsRunning() || c.IsPaused())
```

Comment:
If the container with `--restart=always` exits and pouchd tries to restart it, but the pouchd shutdown by force, such as upgrade action. This container will not be picked up when the pouch restart.

Reply:
Good point, but I think we need more policy to make the situation you mentioned getting the expected result, maybe I can fixed in next PR, WDYT? @fuweid

Reply:
sure